iocore/net/SSLUtils.cc

```diff
@@ -1840,6 +1840,7 @@ SSLAccept(SSL *ssl)
 
       if (ret == SSL_READ_EARLY_DATA_ERROR) {
         Debug("ssl_early_data", "SSL_READ_EARLY_DATA_ERROR");
+        block->free();
```

Comment:
It seems like a common pattern in our code that we call a free() member function to deallocate class instances.  Maybe we should add a utility to itscore that uses std::unique_ptr to call free().  https://godbolt.org/z/G-g7CH It would help avoid accidental leaks like this in future code.

Reply:
I like this idea.

Reply:
Not sure how that's useful here, but maybe I'm missing something. You can't free things here when it goes out of scope, and you don't want to let it sit there if it's not going to be used either. Once used, it'll have to be freed up later, when it's no longer needed (you are collecting this early data as part of the handshake, to be used later by the HttpSM).

Reply:
`netvc->early_data_buf->append_block(block);` would have to change to `netvc->early_data_buf->append_block(block.release());`